chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx

```diff
@@ -0,0 +1,105 @@
+import { PopoverPosition } from '@blueprintjs/core';
+import type { InviteUserDetails } from '@controllers/InviteNewMembers/types';
+import { useStrings } from '@strings';
+import { GetUsersForInvitationOkResponse, useSendInvitationMutation } from '@api/auth';
+import { killEvent } from '@utils';
+import { UserEmail, UserName } from './InviteNewMemberListColumns';
+import css from './InviteNewMemberTable.module.scss';
+
+interface InviteUsersTableViewProps {
+  users: InviteUserDetails[];
+  getUsers: <TPageData>(
+    options?: (RefetchOptions & RefetchQueryFilters<TPageData>) | undefined
+  ) => Promise<QueryObserverResult<GetUsersForInvitationOkResponse, unknown>>;
+}
+
+export default function InviteUsersTableView({ users, getUsers }: InviteUsersTableViewProps): React.ReactElement {
+  const { getString } = useStrings();
+  const envColumns: Column<InviteUserDetails>[] = useMemo(
+    () => [
+      {
+        Header: 'MEMBERS',
+        id: 'username',
+        width: '40%',
+        Cell: UserName
+      },
+      {
+        Header: 'EMAIL',
+        id: 'email',
+        width: '30%',
+        Cell: UserEmail
+      },
+      {
+        Header: '',
+        id: 'threeDotMenu',
+        disableSortBy: true,
+        Cell: ({ row: { original: data } }: { row: Row<InviteUserDetails> }) => {
+          const { projectID } = useParams<{ projectID: string }>();
+          const { showSuccess } = useToaster();
+          const { mutate: sendInvitationMutation, isLoading } = useSendInvitationMutation(
+            {},
+            {
+              onSuccess: () => {
+                showSuccess('Invitation sent successfully');
+                getUsers();
+              }
+            }
+          );
+
+          return (
+            <Layout.Vertical
+              flex={{ justifyContent: 'center', alignItems: 'flex-end' }}
+              onClick={killEvent}
+              width="100%"
+            >
+              <SplitButton
+                text="Invite as"
+                icon="email-inline"
+                variation={ButtonVariation.PRIMARY}
+                loading={isLoading}
+                popoverProps={{
+                  interactionKind: 'click',
+                  usePortal: true,
+                  position: PopoverPosition.BOTTOM_RIGHT
+                }}
+              >
+                <SplitButtonOption
+                  text="Editor"
+                  onClick={() =>
+                    sendInvitationMutation({
+                      body: {
+                        projectID: projectID,
+                        role: 'Editor',
+                        userID: data.id
+                      }
+                    })
+                  }
+                />
+                <SplitButtonOption
+                  text="Viewer"
+                  onClick={() =>
+                    sendInvitationMutation({
+                      body: {
+                        projectID: projectID,
+                        role: 'Viewer',
```

Comment:
enum

Reply:
This is already an enum generated by swagger.
<img width="570" alt="image" src="https://github.com/litmuschaos/litmus/assets/28081206/b7878118-4e10-449c-b6c0-2e8cb9d7fb76">